engines/chewy/detail.h

```diff
@@ -0,0 +1,285 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef CHEWY_DETAIL_H
+#define CHEWY_DETAIL_H
+
+#include "chewy/defines.h"
+#include "chewy/file.h"
+#include "chewy/ngshext.h"
+
+namespace Chewy {
+
+#define SAVE_ROOM 0
+#define RESTORE_ROOM 1
+#define MAX_SOUNDS 3
+
+#define MAX_AUTO_MOV 20
+
+struct RaumInfo {
+	uint8 RoomNr;
+	uint8 BildNr;
+	uint8 AutoMovAnz;
+	uint8 TafLoad;
+	char TafName[14];
+	uint8 ZoomFak;
+	uint8 Dummy;
+
+	bool load(Common::SeekableReadStream *src);
+	static constexpr int SIZE() {
+		return 4 * 1 + 14 + 1 + 1;
+	}
+};
+
+struct RaumAutoMov {
+	int16 X;
+	int16 Y;
+	uint8 SprNr;
+	uint8 dummy;
+
+	bool load(Common::SeekableReadStream *src);
+	static constexpr int SIZE() {
+		return 2 + 2 + 1 + 1;
+	}
+};
+
+#define MAXTAF 10
+#define MAXDETAILS 32
+#define MAX_M_ITEMS 50
+#define MAX_RDI_ROOM 100
+
+struct SprInfo {
+	byte *Image = nullptr;
+	int16 ZEbene = 0;
+	int16 X = 0;
+	int16 Y = 0;
+	int16 X1 = 0;
+	int16 Y1 = 0;
+};
+
+struct sound_def_blk {
+	bool sound_enable[MAX_SOUNDS];
+	int16 sound_index[MAX_SOUNDS];
+	int16 sound_start[MAX_SOUNDS];
+	int16 kanal[MAX_SOUNDS];
+	int16 volume[MAX_SOUNDS];
+	int16 repeats[MAX_SOUNDS];
+	int16 stereo[MAX_SOUNDS];
+
+	bool load(Common::SeekableReadStream *src);
+	static constexpr int SIZE() {
+		return 7 * 2 * MAX_SOUNDS;
+	}
+};
+
+struct ani_detail_info {
+	int16 x;
+	int16 y;
+	uint8 start_flag;
+	uint8 repeat;
+	int16 start_ani;
+	int16 end_ani;
+	int16 ani_count;
+	int16 delay;
+	int16 delay_count;
+	int16 reverse;
+	int16 timer_start;
+	int16 z_ebene;
+	uint8 load_flag;
+	uint8 zoom;
+	sound_def_blk sfx;
+	int16 show_1_phase;
+	int16 phase_nr;
+
+	bool load(Common::SeekableReadStream *src);
+	static constexpr int SIZE() {
```

Comment:
I was wary of simply using sizeof(structures), since I wasn't confident that with byte packing/alignment that the sizes would be as expected on all systems. And I needed the original structure size to be fixed for use in file seek calls to get to the correct offsets to load in the structure data. Is the portability issue the "constexpr"? If so, that could simply be dropped since nothing funky is done with the method results that mandates it's inclusion.